framework/src/base/ComputeNodalKernelsThread.C

```diff
@@ -0,0 +1,85 @@
+/****************************************************************/
```

Comment:
I can agree with that.  One thing I noticed during this work is that we still do NodalBC's _un_-threaded.  That should be fixable.  They will never be super efficient on threads... but they will be better than doing them serially like they are now.  I almost included that fix in this PR... but decided to get this in first and then come back and redo all of the nodal calculations with a threaded node loop base class.

Do you think it's needed for this PR?  Or can we merge it as is and fix this later.  If we can fix this later then I'll put in a ticket for it.